File: backend/app/schemas/ai.py
from pydantic import BaseModel


class StockAlert(BaseModel):
    product_name: str
    current_stock: int
    threshold: int
    recommendation: str


class InventoryAnalysis(BaseModel):
    summary: str
    critical_items: list[StockAlert]
    reorder_recommendations: list[StockAlert]
    insights: list[str]


class SalesSummary(BaseModel):
    headline: str
    total_revenue_insight: str
    top_performer: str
    trend: str  # "growing" | "declining" | "stable"
    key_insights: list[str]
    recommendations: list[str]

File: frontend/src/types/ai.ts
export interface StockAlert {
  product_name: string;
  current_stock: number;
  threshold: number;
  recommendation: string;
}

export interface InventoryAnalysis {
  summary: string;
  critical_items: StockAlert[];
  reorder_recommendations: StockAlert[];
  insights: string[];
}

export interface SalesSummary {
  headline: string;
  total_revenue_insight: string;
  top_performer: string;
  trend: "growing" | "declining" | "stable";
  key_insights: string[];
  recommendations: string[];
}
